chrome/content/zotero/xpcom/plugins.js

```diff
@@ -140,7 +140,10 @@ Zotero.Plugins = new function () {
 			for (let observer of observers) {
 				if (observer[method]) {
 					try {
-						observer[method](params, reason);
+						let maybePromise = observer[method](params, reason);
+						if (maybePromise.then) {
```

Comment:
Needs to be `maybePromise && maybePromise.then` — otherwise a non-async callback without a return value will throw.

Reply:
Right, thanks.